### PR TITLE
feat: impl prover

### PR DESCRIPTION
**Pull request by xander42280**
first commit

### CODE REVIEW

**Comment by xander42280** [comment]
add examples

File: prover/src/provers/prove_prover.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+#[derive(Default)]
+pub struct ProveProver {}
+
+impl ProveProver {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Prover<ProveContext> for ProveProver {
```
Comment (eigmax) [inline prover/src/provers/prove_prover.rs:35]: ProveProver can be renamed to RootProver

File: prover/src/provers/split_prover.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#[derive(Default)]
+pub struct SplitProver {}
+
+impl SplitProver {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl Prover<SplitContext> for SplitProver {
```
Comment (eigmax) [inline prover/src/provers/split_prover.rs:22]: SplitProver is not a prover, you can rename it to Executor, and it doesn't have to implement the Prover interface.

File: prover/src/provers/prove_prover.rs (hunk 1 above)
Comment (xander42280) [reply prover/src/provers/prove_prover.rs]: done

File: prover/src/provers/split_prover.rs (hunk 2 above)
Comment (xander42280) [reply prover/src/provers/split_prover.rs]: done

File: prover/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,59 @@
+[package]
```
[the remark this answers is not in the document]
Comment (xander42280) [reply prover/Cargo.toml:1]: done

File: prover/src/provers/executor.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
+use crate::contexts::SplitContext;
```
Comment (eigmax) [inline prover/src/provers/executor.rs:1]: it's best to move the executor out of prover folder, and put it into executor folder.

**Pull request merged**